backstory/src/test/java/com/google/sps/images/AnnotatedImageTest.java

```diff
@@ -0,0 +1,177 @@
+   *  rawImageData is null.
+   *  labelAnnotations is empty.
+   *  Expects IllegalArgumentException to be thrown.
+  */
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorNullImageData() throws IllegalArgumentException {
+    AnnotatedImage annotatedImageExpectingException = new AnnotatedImage(nullRawImageData, emptyLabelAnnotations);
+  }
+
+  /**
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  rawImageData is empty.
+   *  labelAnnotations is non-empty.
+   *  Expects IllegalArgumentException to be thrown.
+  */
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorEmptyImageData() throws IllegalArgumentException {
+    List<EntityAnnotation> labelAnnotations = new ArrayList<>();
+    labelAnnotations.add(makeMockEntityAnnotationWithDescription("Description")); 
+
+    AnnotatedImage annotatedImageExpectingException = new AnnotatedImage(emptyRawImageData, labelAnnotations);
+  }
+
+  /**
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  rawImageData is non-empty image data.
+   *  labelAnnotations is non-empty.
+  */
+  @Test
+  public void constructorRawImageData() throws IllegalArgumentException, IOException {
+    List<EntityAnnotation> labelAnnotations = new ArrayList<>();
+    labelAnnotations.add(makeMockEntityAnnotationWithDescription("Description"));
+    byte[] rawImageData;
+    try {
+      rawImageData = getBytesFromImageReference(
+        "src/test/java/com/google/sps/images/data/dogRunningOnBeach.jpg", "jpg");
+    } catch (IOException exception) {
+      throw exception;
+    }
+
+    AnnotatedImage actualAnnotatedImage = new AnnotatedImage(rawImageData, labelAnnotations);
+    assertTrue(Arrays.equals(rawImageData, actualAnnotatedImage.getRawImageData()));
+    assertEquals(labelAnnotations, actualAnnotatedImage.getLabelAnnotations());
+  }
+
+  /**
+   * public List<String> getLabelDescriptions();
+   *  labelAnnotations is empty.
+  */
+  @Test
+  public void getLabelDescriptionsEmptyLabelAnnotations() throws IllegalArgumentException, IOException {
+    byte[] rawImageData;
+    try {
+      rawImageData = getBytesFromImageReference(
+        "src/test/java/com/google/sps/images/data/dogRunningOnBeach.jpg", "jpg");
+    } catch (IOException exception) {
+      throw exception;
+    }
+        
+    AnnotatedImage annotatedImageActual = new AnnotatedImage(rawImageData, emptyLabelAnnotations);
+    assertEquals(new ArrayList<>(), annotatedImageActual.getLabelDescriptions());
+  }
+
+  /**
+   * public List<String> getLabelDescriptions();
+   *  labelAnnotations is a non-empty list of EntityAnnotations.
+  */
+  @Test
+  public void getLabelDescriptionsNonEmptyLabelAnnotations() throws IllegalArgumentException, IOException {
+    List<EntityAnnotation> labelAnnotations = new ArrayList<>();
+    labelAnnotations.add(makeMockEntityAnnotationWithDescription("DescriptionOne"));
```

Comment:
Consider making this a local variable, to clearly indicate that the assertion has a dependency on this:

```
String firstAnnotationDescription = @"DescriptionOne";
labelAnnotations.add(makeMockEntityAnnotationWithDescription(firstAnnotationDescription));
...
expectedLabelDescriptions.add(firstAnnotationDescription);
```

Reply:
Okay, I did this.